<launch>

  <arg name="stateEstimationTopic" default="/state_estimation"/>
  <arg name="depthCamPitchOffset" default="0"/>
  <arg name="trackingCamBackward" default="false"/>
  <arg name="trackingCamXOffset" default="0"/>
  <arg name="trackingCamYOffset" default="0"/>
  <arg name="trackingCamZOffset" default="0"/>
  <arg name="trackingCamScale" default="1.0"/>
  <arg name="vehicleX" default="0"/>
  <arg name="vehicleY" default="0"/>
  <arg name="vehicleZ" default="1.0"/>
  <arg name="vehicleYaw" default="0"/>
  <arg name="use_gazebo" default="true"/>
  <arg name="gazebo_gui" default="true"/>
  <arg name="world_name" default="office_simple"/>
  <arg name="realtime_factor" default="1.0"/>

  <include file="$(find ps3joy)/launch/ps3.launch" />

  <include file="$(find control_tuner)/launch/control_tuner.launch" >
    <arg name="stateEstimationTopic" value="$(arg stateEstimationTopic)" />
    <arg name="trackingCamBackward" value="$(arg trackingCamBackward)" />
    <arg name="trackingCamXOffset" value="$(arg trackingCamXOffset)" />
    <arg name="trackingCamYOffset" value="$(arg trackingCamYOffset)" />
    <arg name="trackingCamZOffset" value="$(arg trackingCamZOffset)" />
    <arg name="trackingCamScale" value="$(arg trackingCamScale)" />
  </include>

  <include file="$(find vehicle_simulator)/launch/vehicle_simulator.launch" >
    <arg name="sensorPitch" value="$(arg depthCamPitchOffset)" />
    <arg name="vehicleX" value="$(arg vehicleX)" />
    <arg name="vehicleY" value="$(arg vehicleY)" />
    <arg name="vehicleZ" value="$(arg vehicleZ)" />
    <arg name="vehicleYaw" value="$(arg vehicleYaw)" />
    <arg name="use_gazebo" value="$(arg use_gazebo)" />
    <arg name="gui" value="$(arg gazebo_gui)" />
    <arg name="world_name" value="$(arg world_name)" />
    <arg name="realtime_factor" value="$(arg realtime_factor)" />
  </include>

  <node launch-prefix="nice" pkg="rviz" type="rviz" name="rvizAA" args="-d $(find vehicle_simulator)/rviz/vehicle_simulator_gazebo.rviz" respawn="true"/>

  <node pkg="tf" type="static_transform_publisher" name="vehicleTransPublisher" args="0 0 0 -1.5707963 0 -1.5707963 /vehicle /rgbd_camera 1000"/>

</launch>
